file-store/src/main/java/com/expedia/blobs/stores/io/FileStore.java

```diff
@@ -82,9 +87,61 @@ protected void storeInternal(Blob blob) {
             final Map<String, String> metadata = gson.fromJson(meta, mapType);
             final byte[] data = FileUtils.readFileToByteArray(new File(blobPath));
             return Optional.of(new SimpleBlob(key, metadata, data));
-        }
-        catch (IOException e) {
+        } catch (IOException e) {
             throw new BlobReadWriteException(e);
         }
     }
+
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    /**
+     * Builds the {@link FileStore} with options
+     */
+
+    public static class Builder {
+        private final File directory;
+        private int threadPoolSize;
+        private int shutdownWaitInSeconds;
+        private boolean manualShutdown;
+
+        public Builder(String directory) {
+            this(new File(directory));
+        }
+
+        public Builder(File directory) {
+            this.directory = directory;
+            this.threadPoolSize = Runtime.getRuntime().availableProcessors();
+            this.shutdownWaitInSeconds = 60;
+            this.manualShutdown = false;
+        }
+
+        public Builder withThreadPoolSize(int threadPoolSize) {
+            this.threadPoolSize = threadPoolSize;
+            return this;
+        }
+
+        public Builder withShutdownWaitInSeconds(int shutdownWaitInSeconds) {
+            this.shutdownWaitInSeconds = shutdownWaitInSeconds;
+            return this;
+        }
+
+        public Builder withManualShutdown() {
```

Comment:
Can we say `disableAutoShutdown` instead and set `this.closeOnShutdown = false` ?   this will mean the default is enabled.  This will allow us to code  ``` if (builder.closeOnShutdown) { //add shutdown hook }```

I am not sure why someone would want us to not close.   I am fine with a disable close method if someone really wants to

Reply:
@vaibhavsawhney this is not resolved yet